Add a pre-commit hook for developers

IRDonch opened the pull request:
<!-- Contributing guide: https://github.com/openvinotoolkit/datumaro/blob/develop/CONTRIBUTING.md -->

### Summary
It will help to remind people to format their code before committing.

This also establishes `dev` as a directory for developer-oriented scripts.

<!--
Resolves #111 and #222.
Depends on #1000 (for series of dependent commits).

This PR introduces this capability to make the project better in this and that.

- Added this feature
- Removed that feature
- Fixed the problem #1234
-->

### How to test
<!-- Describe the testing procedure for reviewers, if changes are
not fully covered by unit tests or manual testing can be complicated. -->

### Checklist
<!-- Put an 'x' in all the boxes that apply -->
- [x] I submit my changes into the `develop` branch
- [ ] I have added description of my changes into [CHANGELOG](https://github.com/openvinotoolkit/datumaro/blob/develop/CHANGELOG.md)
- [ ] I have updated the [documentation](
  https://github.com/openvinotoolkit/datumaro/tree/develop/docs) accordingly
- [ ] I have added tests to cover my changes
- [ ] I have [linked related issues](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/openvinotoolkit/datumaro/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.
- [x] I have updated the license header for each file (see an example below)

```python
# Copyright (C) 2021 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

Commit:
Add a pre-commit hook for developers

It will help to remind people to format their code before committing.

This also establishes `dev` as a directory for developer-oriented scripts.

IRDonch commented:
CC @kirill-sizov @yasakova-anastasia

zhiltsov-max commented on dev/pre-commit.py:
```diff
@@ -0,0 +1,44 @@
+
+To install, make sure isort and black are installed on your system, then run:
+
+$ ln -sv ../../dev/pre-commit.py ./.git/hooks/pre-commit
```
Probably, it would be good to mention this in Developer guide.

IRDonch replied:
I added a mention in `CONTRIBUTING.md`, since that's where the coding style is described.

Commit:
dev/pre-commit.py: suppress bandit warnings

All inputs to subprocess come from our own repository, so there should not be
any security problems here.

IRDonch commented:
I updated the hook so that only changed files are scanned. This makes it much faster.
